File: 0101-0200/0169.Majority_Element.java
//0ms







class Solution {
    public static int majorityElement(int[] nums) {
        return helper(nums,0,nums[0]);
    }static int helper(int[] nums, int si, int ref){
        int c=0;
        for(int i=si;i<nums.length;i++){
            if(nums[i]==ref)
                c++;
            else
                c--;
            if(c==-1)
                return helper(nums,i,nums[i]);
        }return ref;
    }
}







//1ms





class Solution {
    public int majorityElement(int[] nums) {
       int element=0;// store majority element
       int count=0;  //counter

// Traverse the array to find the majority element
       for(int i=0; i<nums.length; i++){
        if(count==0){
// Adopt a new element
            element= nums[i];
        }
        if(element== nums[i]) count++;   //increment if element is same
        else count--;  //decrement id different, to adopt new element
       }

        // Since majority element is guaranteed to exist,
       return element;
    }
}  

File: 0101-0200/0169.Majority_Element.py
#0ms





class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        a=set(nums)
        maxy=0
        ele=0
        for i in a:
            rep=nums.count(i)
            if rep>=maxy:
                maxy=rep
                ele=i
        return ele





#1ms







class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        median=int(len(nums)/2)
        return(nums[median])
